File: backend/Models/Order.cs

using Stripe;

namespace DotnetReactShop.Models
{
    public class Order
    {
        public int Id { get; set; } // primary key

        public string UserId { get; set; } // ref to user (if possible?)
        public ApplicationUser? User { get; set; } // optional nav property

        public string ShippingName { get; set; }
        public string ShippingAddress { get; set; }
        public string ShippingCity { get; set; }
        public string ShippingPostalCode { get; set; }
        public string ShippingCountry { get; set; }

        public decimal Total { get; set; }
        public DateTime OrderDate { get; set; }

        public List<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

        public void CalculateTotal()
        {
            Total = OrderItems.Sum(item => item.UnitPrice * item.Quantity);
        }
    }
}

File: frontend/src/types/Order.ts
export interface OrderItemRaw {
  id: number;         // primary key of the OrderItem record
  orderId: number;    // FK back to the parent order
  productId: number;  // FK to Product (we’ll just show productId for now)
  quantity: number;
  unitPrice: number;
}

export interface OrderRaw {
  id: number;                // primary key of the order
  userId: string;            // the retailer’s user ID or email
  orderDate: string;         // ISO date string; e.g. "2025-06-01T00:00:00"
  shippingName: string;
  shippingAddress: string;
  shippingCity: string;
  shippingPostalCode: string;
  shippingCountry: string;
  orderItems: OrderItemRaw[];  // array of line items (no product name yet)
}